flow/scene_update_context.cc

```diff
@@ -299,12 +299,16 @@ SceneUpdateContext::Frame::Frame(SceneUpdateContext& context,
       color_(color),
       paint_bounds_(SkRect::MakeEmpty()),
       layer_(layer) {
+  if (depth > -1 && world_elevation > depth) {
+    // TODO(mklim): Deal with bounds overflow correctly.
```

Comment:
Is this TODO intended? i.e., is this still an incorrect handling?

Reply:
It's correct and fixes the potential clipping, but ideally we wouldn't be just clamping in all cases and there'd be some alternative options for the developers here (like normalization) since clamping could end up looking strange.

Reply:
I see. If this is not an urgent fix, can we please add tests to verify the change made here?

I understand that currently there might be difficulties in writing Fuchsia unit tests inside Flutter engine repo. But I think now is a good time to solve such difficulties as we're facing more and more bug reports from Fuchsia, and @chinmaygarde is moving Flutter runner to the engine repo.

Chinmay and Ariana (from Fuchsia team) suggested AEMU. Do you think that we can add some unit tests here?

Reply:
I'll investigate! This isn't immediately critical, but in general this is important to prevent unexpected clipping so I also wouldn't want this to be blocked for long either.